lib/token-store/index.js

```diff
@@ -540,4 +541,71 @@ module.exports = class TokenStore {
             padding
         ).getAllTokens();
     }
+
+    /**
+     * Gets all comments before the given node or token.
+     * @param {ASTNode|token} nodeOrToken The AST node or token to get the comments for.
+     * @returns {Object} The list of comments indexed by their position.
+     * @public
+     */
+    getCommentsBefore(nodeOrToken) {
+        const tokens = [];
+        const cursor = createCursorWithCount(
+            cursors.backward,
+            this[TOKENS],
+            this[COMMENTS],
+            this[INDEX_MAP],
+            -1,
+            nodeOrToken.range[0],
+            { includeComments: true }
+        );
+        let currentToken = cursor.getOneToken();
+
+        while (currentToken && astUtils.isCommentToken(currentToken)) {
+            tokens.push(currentToken);
+            currentToken = cursor.getOneToken();
+        }
+
+        return tokens.reverse();
+    }
+
+    /**
+     * Gets all comments after the given node or token.
+     * @param {ASTNode|token} nodeOrToken The AST node or token to get the comments for.
+     * @returns {Object} The list of comments indexed by their position.
+     * @public
+     */
+    getCommentsAfter(nodeOrToken) {
+        const tokens = [];
+        const cursor = createCursorWithCount(
```

Comment:
All the logic inside `getCommentsAfter` and `getCommentsBefore` is 99% same. Why not just extract that into a common function?